buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/PrecommitTasks.groovy

```diff
@@ -130,27 +130,29 @@ class PrecommitTasks {
         }
 
         Task checkstyleTask = project.tasks.create('checkstyle')
-        // Apply the checkstyle plugin to create `checkstyleMain` and `checkstyleTest`. It only
-        // creates them if there is main or test code to check and it makes `check` depend
-        // on them. But we want `precommit` to depend on `checkstyle` which depends on them so
-        // we have to swap them.
-        project.pluginManager.apply('checkstyle')
-        project.checkstyle {
-            config = project.resources.text.fromFile(checkstyleConf, 'UTF-8')
-            configProperties = [
-                suppressions: checkstyleSuppressions
-            ]
-            toolVersion = 7.5
-        }
-        for (String taskName : ['checkstyleMain', 'checkstyleTest']) {
-            Task task = project.tasks.findByName(taskName)
-            if (task != null) {
-                project.tasks['check'].dependsOn.remove(task)
-                checkstyleTask.dependsOn(task)
-                task.dependsOn(copyCheckstyleConf)
-                task.inputs.file(checkstyleSuppressions)
-                task.reports {
-                    html.enabled false
+        project.afterEvaluate {
```

Comment:
I don't think we need an afterEvaluate. Instead, I think we can use:

```project.tasks.withType(Checkstyle) { Checkstyle checkstyleTask ->
  dependsOn(copyCheckstyleConf)
}```

Reply:
I might be misunderstanding but I do not think that works; the problem here, and the reason for the `afterEvaluate` is that the checkstyleJava9 task is not defined when we create the precommit tasks. The source set for Java 9 is added later. This is why I do this in an `afterEvaluate`, to ensure that the Java 9 source set is added before the checkstyle tasks are configured.

Reply:
Many of the methods that gradle has on collections, like `withType`, operate not only on the current objects in that collection, but also on future objects added. So the closure applied to withType here will also be applied to the java9 checkstyle when it is later added.